pkg/ttl/cache/split_test.go

```diff
@@ -825,3 +928,182 @@ func TestGetNextIntHandle(t *testing.T) {
+			d:   types.NewIntDatum(math.MaxInt64),
+		},
+		{
+			key: encode(tblID, types.NewIntDatum(123), types.NewIntDatum(0)),
+			d:   types.NewIntDatum(124),
+		},
+		{
+			key: encode(tblID, types.NewIntDatum(-123), types.NewIntDatum(0)),
+			d:   types.NewIntDatum(-122),
+		},
+		{
+			key: encode(tblID, types.NewIntDatum(math.MinInt64), types.NewIntDatum(0)),
+			d:   types.NewIntDatum(math.MinInt64 + 1),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(0)),
+			d:   types.NewUintDatum(0),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(1)),
+			d:   types.NewUintDatum(1),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(1024)),
+			d:   types.NewUintDatum(1024),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(math.MaxInt64)),
+			d:   types.NewUintDatum(math.MaxInt64),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(math.MaxInt64+1)),
+			d:   types.NewUintDatum(math.MaxInt64 + 1),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(math.MaxUint64)),
+			d:   types.NewUintDatum(math.MaxUint64),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(math.MaxUint64))[:fixedLen-1],
+			d:   types.NewUintDatum(math.MaxUint64 - 0xFF),
+		},
+
+		{
+			key: encode(tblID, types.NewUintDatum(math.MaxUint64), types.NewIntDatum(0)),
+			d:   nullDatum,
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(math.MaxUint64-1), types.NewIntDatum(0)),
+			d:   types.NewUintDatum(math.MaxUint64),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(123), types.NewIntDatum(0)),
+			d:   types.NewUintDatum(124),
+		},
+		{
+			key: encode(tblID, types.NewUintDatum(0), types.NewIntDatum(0)),
+			d:   types.NewUintDatum(1),
+		},
+		{
+			key: []byte{},
+			d:   types.NewIntDatum(math.MinInt64),
+		},
+		{
+			key: []byte{},
+			d:   types.NewUintDatum(0),
+		},
+		{
+			key: tablecodec.GenTableRecordPrefix(tblID),
+			d:   types.NewIntDatum(math.MinInt64),
+		},
+		{
+			key: tablecodec.GenTableRecordPrefix(tblID),
+			d:   types.NewUintDatum(0),
+		},
+		{
+			// 3 is encoded intFlag
+			key: append(tablecodec.GenTableRecordPrefix(tblID), []byte{3}...),
+			d:   types.NewIntDatum(math.MinInt64),
+		},
+		{
+			// 3 is encoded intFlag
+			key: append(tablecodec.GenTableRecordPrefix(tblID), []byte{3}...),
+			d:   types.NewUintDatum(0),
+		},
+		{
+			// 4 is encoded uintFlag
+			key: append(tablecodec.GenTableRecordPrefix(tblID), []byte{4}...),
+			d:   nullDatum,
+		},
+		{
+			// 4 is encoded uintFlag
+			key: append(tablecodec.GenTableRecordPrefix(tblID), []byte{4}...),
+			d:   types.NewUintDatum(0),
+		},
+		{
+			// 5
+			key: append(tablecodec.GenTableRecordPrefix(tblID), []byte{5}...),
+			d:   nullDatum,
+		},
+		{
+			// 5
+			key:      append(tablecodec.GenTableRecordPrefix(tblID), []byte{5}...),
+			d:        nullDatum,
+			unsigned: true,
+		},
+	}
+
+	for _, c := range cases {
+		d := cache.GetNextIntDatumFromCommonHandle(c.key, tablecodec.GenTableRecordPrefix(tblID), c.d.Kind() == types.KindUint64 || c.unsigned)
```

Comment:
Use `c.unsigned` instead of `c.d.Kind() == types.KindUint64` and modify cases will easier to read code.